### PR TITLE
[AN-4527][AN-4528][AN-4529][AN-4530][AN-VIDEO][AN-4576] More ephemera…

### PR DESCRIPTION
**Pull request by hakonbo**
…l stuff

### CODE REVIEW

**Comment by hakonbo** [comment]
build again

**Comment by mykola-mokhnach** [comment]
App crashed when I was trying to close participants view of group conversation:
E/AndroidRuntime(17019): FATAL EXCEPTION: main
E/AndroidRuntime(17019): Process: com.wire.x, PID: 17019
E/AndroidRuntime(17019): java.lang.NullPointerException: Attempt to invoke interface method 'com.waz.api.IConversation$Type com.waz.api.IConversation.getType()' on a null object reference
E/AndroidRuntime(17019): 	at com.waz.zclient.ui.cursor.CursorLayout.showSendButton(CursorLayout.java:423)
E/AndroidRuntime(17019): 	at com.waz.zclient.pages.main.conversation.ConversationFragment.onEditTextHasChanged$4f708078(ConversationFragment.java:1171)
E/AndroidRuntime(17019): 	at com.waz.zclient.ui.cursor.CursorLayout.onTextChanged(CursorLayout.java:333)
E/AndroidRuntime(17019): 	at android.widget.TextView.sendOnTextChanged(TextView.java:7722)
E/AndroidRuntime(17019): 	at android.widget.TextView.setText(TextView.java:4099)
E/AndroidRuntime(17019): 	at android.widget.TextView.setText(TextView.java:3954)
E/AndroidRuntime(17019): 	at android.widget.EditText.setText(EditText.java:85)
E/AndroidRuntime(17019): 	at android.widget.TextView.setText(TextView.java:3929)
E/AndroidRuntime(17019): 	at com.waz.zclient.ui.cursor.CursorLayout.setText(CursorLayout.java:432)
E/AndroidRuntime(17019): 	at com.waz.zclient.pages.main.conversation.ConversationFragment.resetCursor(ConversationFragment.java:1542)
E/AndroidRuntime(17019): 	at com.waz.zclient.pages.main.conversation.ConversationFragment.access$1900(ConversationFragment.java:186)
E/AndroidRuntime(17019): 	at com.waz.zclient.pages.main.conversation.ConversationFragment$14.run(ConversationFragment.java:947)
E/AndroidRuntime(17019): 	at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime(17019): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(17019): 	at android.os.Looper.loop(Looper.java:211)
E/AndroidRuntime(17019): 	at android.app.ActivityThread.main(ActivityThread.java:5389)
E/AndroidRuntime(17019): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(17019): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(17019): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1020)
E/AndroidRuntime(17019): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:815)

**Comment by mykola-mokhnach** [comment]
"Times Message" placeholder is shown even if Message timer is set to Off:
screen

**Comment by mykola-mokhnach** [comment]
The glass-hour icon is supposed to be changed to timeout number icon after the timeout value is selected

**Comment by mykola-mokhnach** [comment]
Ephemeral timer is not shown for pings. Also, hot ping is immediately replaced with expired placeholder if the previous ping message has been expired already

**Comment by mykola-mokhnach** [comment]
Ephemeral timer is not shown for Giphy messages after I send them

**Comment by vova-lohika** [comment]
ghost message is not removed if received side client was in background.

**Comment by mykola-mokhnach** [comment]
Tapping an image/sketch which is just sent as ephemeral message has no effect. It does not show neither fullscreen nor sketch buttons. But after this message is expired it is possible to tap it and these two buttons appear on top of orange placeholder. It's even possible to tap Sketch button, which sets the picture (non obfuscated) as the background.

**Comment by mykola-mokhnach** [comment]
Preview generation for Soundcloud links is broken

**Comment by vova-lohika** [comment]
received side see empty messages if they are ephemera:
- pictures
- gif
- sketch

**Comment by mykola-mokhnach** [comment]
The color of obfuscated placeholder for youtube previews differs from the color of obfuscated placeholders for other message types.

**Comment by vova-lohika** [comment]
Receiver don't see timer to ephemera location

**Comment by vova-lohika** [comment]
Ping ghost message is not removed on sender side

File: app/src/main/java/com/waz/zclient/pages/extendedcursor/ephemeral/EphemeralLayout.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        numberPicker.setMaxValue(ephemeralExpirationsValues.length - 1);
+        String[] values = new String[ephemeralExpirationsValues.length];
+        for (int i = 0; i < values.length; i++) {
+            values[i] = getDisplayName(ephemeralExpirationsValues[i]);
+        }
+        numberPicker.setDisplayedValues(values);
+        numberPicker.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Timber.w("Vova EphemeralLayout onClick selected=%s, callback=%s", ephemeralExpirationsValues[numberPicker.getValue()].name(), callback);
```
Comment (shumengye) [inline app/src/main/java/com/waz/zclient/pages/extendedcursor/ephemeral/EphemeralLayout.java]: remove

Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/pages/extendedcursor/ephemeral/EphemeralLayout.java]: Nah, not yet. He's debugging something.. Will remove eventually.

**Comment by mykola-mokhnach** [comment]
1. Send a Soundcloud message with 15 seconds expiration timeout
2. Tap play button
3. Wait until timeout expires

AR:
the playback is still in progress, but the preview itself is not available anymore and it is not possible to stop the playback

ER:
The playback should stop automatically as soon as the message is expired

**Comment by mykola-mokhnach** [comment]
Clicking obfuscated shared location messages still opens Google Maps application and shows a location

**Comment by olivereitalu** [comment]
- Hard to select time
- Instead of hourglass there should be a time indicator
- Once something is written there should be send button of the same orange color
- Limit the selection of timers to 5s, 15s and 1m
- Selector background and lines
- Line above the input field never goes away
- Hint above the selector should say “Set a timer for timed messages”

**Comment by hakonbo** [comment]
build

**Comment by hakonbo** [comment]
build again

**Comment by vova-lohika** [comment]
No notification for ephemera file sharing

**Comment by vova-lohika** [comment]
Failed to send ephemera message is resent as rubbish

**Pull request merged**